packages/wallet/src/wallet.ts

```diff
@@ -31,91 +28,176 @@ export interface WalletApp {
-            let { sender, value } = parseEtherDeposit(data.payload);
-            const wallet = this.wallets[sender] ?? { ether: 0n, erc20: {} };
-            wallet.ether += value;
-            this.wallets[sender] = wallet;
-            return "accept";
-        } else if (isERC20Deposit(data)) {
-            let { success, token, sender, amount } = parseERC20Deposit(
-                data.payload,
-            );
+    public balanceOfERC721(
+        address: string | Address,
+        owner: string | Address,
+    ): bigint {
+        const handler = TokenHandler.getInstance();
+        return handler.erc721.balanceOf({
+            address,
+            getWallet: this.getWalletOrNew,
+            owner,
+        });
+    }
 
-            if (success) {
-                const wallet = this.wallets[sender] ?? { ether: 0n, erc20: {} };
-                wallet.erc20[token] = wallet.erc20[token]
-                    ? wallet.erc20[token] + amount
-                    : amount;
-                this.wallets[sender] = wallet;
-            }
-            return "accept";
-        } else if (
-            getAddress(data.metadata.msg_sender) === dAppAddressRelayAddress
-        ) {
-            this.dapp = getAddress(data.payload);
-            return "accept";
+    public balanceOfERC1155(
+        addresses: string | string[],
+        tokenIds: bigint | bigint[],
+        owner: string | Address,
+    ): bigint | bigint[] {
+        const handler = TokenHandler.getInstance();
+
+        if (!Array.isArray(addresses) && !Array.isArray(tokenIds)) {
+            const address = addresses;
+            const tokenId = tokenIds;
+
+            return handler.erc1155Single.balanceOf({
+                address,
+                tokenId,
+                owner,
+                getWallet: this.getWalletOrNew,
+            });
         }
-        return "reject";
-    };
 
-    transferEther(from: string, to: string, amount: bigint): void {
-        // normalize addresses
-        if (isAddress(from)) {
-            from = getAddress(from);
+        if (!Array.isArray(addresses)) {
+            addresses = [addresses];
         }
-        if (isAddress(to)) {
-            to = getAddress(to);
+
+        if (!Array.isArray(tokenIds)) {
+            tokenIds = [tokenIds];
         }
 
-        const walletFrom = this.wallets[from] ?? { ether: 0n, erc20: {} };
-        const walletTo = this.wallets[to] ?? { ether: 0n, erc20: {} };
+        return handler.erc1155Batch.balanceOf({
+            addresses,
+            tokenIds,
+            owner,
+            getWallet: this.getWalletOrNew,
+        });
+    }
 
-        if (walletFrom.ether < amount) {
-            throw new Error(`insufficient balance of user ${from}`);
+    public handler: AdvanceRequestHandler = async (data) => {
+        try {
+            console.log("Wallet handler...", inspect(data, { depth: null }));
+
+            const tokenHandler = TokenHandler.getInstance();
```

Comment:
this `TokenHandler.getInstance` is weird. Why?

Reply:
The token or contract `Tokenhandler` is a singleton with a facade for each of the items: ERC20, Ether, ERC1155 (Single/Batch), ERC721.
It is useful in using the `handler` method where he find which token or contract should be used and executed.
In other methods, it is called the instance of this `Tokenhandler`.

Reply:
I’d rather not use this pattern and just use the handler pattern that deroll itself uses.